File: app/src/main/java/com/merino/ddfilms/DDFilmsApplication.java
package com.merino.ddfilms;

import android.app.Application;
import android.content.SharedPreferences;
import androidx.appcompat.app.AppCompatDelegate;

public class DDFilmsApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        SharedPreferences preferences = getSharedPreferences("Preferences", MODE_PRIVATE);
        String themeMode = preferences.getString("theme_mode", "system");
        applyTheme(themeMode);
    }

    public static void applyTheme(String themeMode) {
        switch (themeMode) {
            case "light":
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
                break;
            case "dark":
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
                break;
            case "system":
            default:
                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
                break;
        }
    }
}

File: app/src/main/java/com/merino/ddfilms/DDFilmsApplication.kt
package com.merino.ddfilms

import android.app.Application
import android.content.Context
import androidx.appcompat.app.AppCompatDelegate

class DDFilmsApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        val preferences = getSharedPreferences("Preferences", Context.MODE_PRIVATE)
        val themeMode = preferences.getString("theme_mode", "system") ?: "system"
        applyTheme(themeMode)
    }

    companion object {
        @JvmStatic
        fun applyTheme(themeMode: String) {
            when (themeMode) {
                "light" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                "dark" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                "system" -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
                else -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
            }
        }
    }
}
